src/App.tsx

```diff
@@ -15,48 +16,136 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
+
+export const App = () => {
+  const [sortType, setType] = useState(SortType.NONE);
+  const [isReversed, setReverse] = useState(false);
+
+  const alphabeticSort = () => {
+    setType(SortType.ALPHABET);
+  };
+
+  const lengthSort = () => {
+    setType(SortType.LENGTH);
+  };
+
+  const resetSort = () => {
+    setType(SortType.NONE);
+    setReverse(false);
+  };
+
+  const reversed = () => {
+    setReverse(true);
+  };
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
```

Comment:
in such cases, it's better to format code the following way
```suggestion
          className={cn('button is-info', {
            'is-light': sortType !== SortType.ALPHABET,
          })}
```